src/routes/profile.router.js

```diff
@@ -0,0 +1,172 @@
+const { User } = require('../models/User.model');
+const { ApiError } = require('../exceptions/api.error');
+const { v4: uuidv4 } = require('uuid');
+const emailService = require('./email.service');
+const userService = require('./user.service');
+const bcrypt = require('bcrypt');
+const jwtService = require('./jwt.service');
+const tokenService = require('./token.service');
+const { checkRequired } = require('../utils/checkRequired');
+const { validateEmail } = require('../utils/validateEmail');
+const { validatePassword } = require('../utils/validatePassword');
+
+const register = async ({ name, email, password }) => {
+  const errors = {
+    name: checkRequired(name, 'name'),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.name || errors.password) {
+    throw ApiError.BadRequest('bad request', errors);
+  }
+
+  if (await userService.getByEmail(email)) {
+    throw ApiError.BadRequest('bad request', {
+      userExists: 'user already exists',
+    });
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+  const activationToken = uuidv4();
+  const newUser = await User.create({
+    name,
+    email,
+    password: hashedPass,
+    activationToken,
+  });
+
+  await emailService.sendActivationEmail(email, activationToken);
+
+  return newUser;
+};
+
+const activate = async (activationToken) => {
+  const user = await userService.getByActivationToken(activationToken);
+
+  if (!user) {
+    throw ApiError.notFound();
```

Comment:
The method 'notFound' should be 'NotFound' to match the naming convention used in the ApiError class.